### PR TITLE
Functional tests config

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-683

### Change description ###
- Add s2s environment variables for tests in Jenkins files.
- Rename test s2s secret name for sample app tests (conflicts with the existing key vault property of processor)
- Allow ocr data validation requests from sample app tests
- The above changes should be deployed in aat to run the functional tests successfully.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -23,4 +23,4 @@ idam:
     secret: ${S2S_SECRET:AAAAAAAAAAAAAAAA}
     name:   ${S2S_NAME:bulk_scan_ccd_sample_app}
 
-allowed-services: bulk_scan_processor,bulk_scan_processor_test
+allowed-services: bulk_scan_processor,bulk_scan_processor_test,bulk_scan_sample_app_tests
```
Comment (nitinprabhu) [inline src/main/resources/application.yaml]: I think you can remove processor tests ?

File: charts/bulk-scan-sample-app/values.template.yaml [hunk 2]
```diff
@@ -2,9 +2,10 @@ java:
...
   environment:
     image: ${IMAGE_NAME}
+    S2S_NAME: "bulk_scan_ccd_sample_app"
 
   keyVaults:
     "s2s":
       resourceGroup: rpe-service-auth-provider
       secrets:
-        - microservicekey-bulk-scan-orchestrator
+        - microservicekey-bulk-scan-sample-app-tests
```
Comment (nitinprabhu) [inline charts/bulk-scan-sample-app/values.template.yaml]: Want to bump chart version ?

Comment (aliveni) [reply charts/bulk-scan-sample-app/values.template.yaml]: done

File: src/main/resources/application.yaml (hunk 1 above)
Comment (aliveni) [reply src/main/resources/application.yaml]: removed processor tests

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -11,7 +11,7 @@ def channel = '#bsp-build-notices'
 GradleBuilder builder = new GradleBuilder(this, product)
 
 List<LinkedHashMap<String, Object>> secrets = [
-  secret('test-s2s-secret', 'TEST_S2S_SECRET')
+  secret('test-sample-app-s2s-secret', 'TEST_S2S_SECRET')
```
Comment (rkondratowicz) [inline Jenkinsfile_CNP]: is this an s2s secret for service called `bulk-scan-sample-app-tests`?

Comment (aliveni) [reply Jenkinsfile_CNP]: yeah

Comment (aliveni) [reply Jenkinsfile_CNP]: we already used `test-s2s-secret` name for `bulk-scan-processor-tests`. so I had to rename it

Comment (rkondratowicz) [reply Jenkinsfile_CNP]: maybe `bulk-scan-sample-app-tests-s2s-secret` for clarity?

Comment (aliveni) [reply Jenkinsfile_CNP]: renamed. Also, deleted the old secret from vault

**Pull request merged**